### ISSUE TITLE
Issues from discussion Friday 17th

### ISSUE THREAD

**Issue by vincecr0ft** [matthewfeickert/fellowship-project#3]
- [ ] clean up code to investigate why TF fit takes so long.
    - [x] Try alter required accuracy
    - [x] Remove superflous parameters
- [ ] Pass variable numbers of parameters to the session rather than hardcoded Mu and Sigma
- [x] Use built in TensorFlow minimiser options
- [ ] make a new list next week (20-25 nov)

Longer term:
- Modularise
- Pass histograms instead of a simple Gaussian
- Make variable histogram model. 

**Goal:** first implementation prior to 120fb-1 workshop. ~2 weeks

**Comment by matthewfeickert** [comment]
I think the reason the MLE fit is taking so long is that I didn't write it with a very good graph structure in mind. I am trying to rewrite the function now so that almost all of the computation is predefined in the graph before initialization, and then the session just needs to execute the graph in the while loop. Additionally, the learning rate of the optimizer being currently used ([`tf.train.AdamOptimizer`](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer))  seems to have a very large effect on the fit time (that is `learning_rate=0.001` will take well over 20 seconds to fit).
